src/main.c

```diff
@@ -0,0 +1,441 @@
+#include <ctype.h>
+#include <dirent.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <linux/nvme_ioctl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+struct nvme_lbaf
+{
+    __le16 ms;
+    __u8 ds;
+    __u8 rp;
+};
+
+struct nvme_id_ns
```

Comment:
We can but then there is the trick of finding the kernel headers properly which I wanted to avoid for the time being. 

For now, I just moved them into nvme.h and we can reconsider after we do some OS packaging and can experiment.

Reply:
Sure, this is reasonable